File: src/main/java/com/flexpoker/game/query/repository/GameListRepository.java
package com.flexpoker.game.query.repository;

import java.util.List;
import java.util.UUID;

import com.flexpoker.game.query.dto.GameInListDTO;
import com.flexpoker.game.query.dto.GameStage;

public interface GameListRepository {

    void saveNew(GameInListDTO gameInListDTO);

    List<GameInListDTO> fetchAll();

    void incrementRegisteredPlayers(UUID aggregateId);

    String fetchGameName(UUID aggregateId);

    void changeGameStage(UUID aggregateId, GameStage gameStage);

}

File: src/main/kotlin/com/flexpoker/game/query/repository/GameListRepository.kt
package com.flexpoker.game.query.repository

import com.flexpoker.game.query.dto.GameInListDTO
import com.flexpoker.game.query.dto.GameStage
import java.util.UUID

interface GameListRepository {
    fun saveNew(gameInListDTO: GameInListDTO)
    fun fetchAll(): List<GameInListDTO>
    fun incrementRegisteredPlayers(aggregateId: UUID)
    fun fetchGameName(aggregateId: UUID): String
    fun changeGameStage(aggregateId: UUID, gameStage: GameStage)
}